src/main/java/seedu/address/ui/TaskTabPanel.java

```diff
@@ -61,60 +69,93 @@
     private AnchorPane todayTasksListPanelPlaceholder;
 
     @FXML
-    private TabPane taskTabPanel;
+    private TabPane taskTabPane;
 
     /**
      * @param placeholder The AnchorPane where the taskTabPanel must be inserted
      */
-    public TaskTabPanel(AnchorPane placeholder, ObservableList<ReadOnlyTask> taskList) {
+    public TaskTabPanel(AnchorPane placeholder, Logic logic) {
```

Comment:
I personally think you are creating too much coupling if you pass the entire logic in though. So it's up to you if you want to do this.